helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroRecordUtil.java

```diff
@@ -77,4 +87,295 @@ private static IndexedRecord supplementDefaults(IndexedRecord record, boolean th
     }
     return record;
   }
+
+  /**
+   * converts a given {@link GenericData.Record} into an instance of the "equivalent"
+   * {@link SpecificRecord} (SR) class. Can optionally reuse a given SR instance as output,
+   * otherwise SR class will be looked up on the current thread's classpath by fullname
+   * (or optionally by aliases)
+   * @param input generic record to convert from. required
+   * @param outputReuse specific record to convert into. optional. if provided must be of the correct
+   *                    fullname (matching input or any of input's aliases, depending on config)
+   * @param config configuration for the operation
+   * @param <T> type of the SR class
+   * @return a SR converted from the input
+   */
+  public static <T extends SpecificRecord> T genericRecordToSpecificRecord(
+          GenericRecord input,
+          T outputReuse,
+          RecordConversionConfig config
+  ) {
+    if (input == null) {
+      throw new IllegalArgumentException("input required");
+    }
+    if (config == null) {
+      throw new IllegalArgumentException("config required");
+    }
+    Schema inputSchema = input.getSchema();
+    Schema outputSchema;
+    ClassLoader cl;
+    T outputRecord;
+    if (outputReuse == null) {
+      //use context loader
+      Class<T> srClass;
+      cl = Thread.currentThread().getContextClassLoader();
+      //look up SR class by fullname and possibly aliases
+      if (config.isUseAliasesOnNamedTypes()) {
+        //TODO - do we need to convert simple aliases to fullnames?
```

Comment:
I'm assuming that could happen when the namespace of a record changes, right?

Reply:
yes. type aliases arer used if the destination class isnt found by the "writer" fullname